### PR TITLE
STRIPE change to use source instead of card

### PR DESCRIPTION
**Pull request by ptibogxiv**
use of more Gateways (3d secure, sepa direct debit…)  and add payment request API ( google pay, apple pay, Microsoft pay…)

### CODE REVIEW

**Comment by ptibogxiv** [comment]
saving and pay function have some bugs, fix in progress

**Comment by ptibogxiv** [comment]
Bugs for saving and payment are fixed :-)

File: htdocs/public/payment/newpayment.php [hunk 1]
```diff
@@ -1542,97 +1548,3473 @@ class="stripe-button"
...
+      jQuery('#buttontopay').hide();
+      jQuery('#hourglasstopay').show();
+      console.log("submit");
+      form.submit();
+    }
 
-	    <?php
-		print '</script>';
+<?php
+/* Copyright (C) 2001-2002	Rodolphe Quiedeville	<rodolphe@quiedeville.org>
```
Comment (eldy) [inline htdocs/public/payment/newpayment.php]: Header of a php file in a middle of the php file ?

Comment (ptibogxiv) [reply htdocs/public/payment/newpayment.php]: @eldy fix of my bad copy/paste it's done

**Comment by ptibogxiv** [comment]
@eldy form with source  and apple/google/Microsoft pay is functionnal. i Don't find the perfect internal css class in dolibarr for a better look/ergonomy.  why Don't rewrite new payment in a one screen form with all Gateway in the same list as on my form below
capture

**Comment by eldy** [comment]
There was so many change since this PR, that there was too much conflict that i failed to solve. Your help is welcome because being able to manage other stripe payment mode would be fantastic !

**Comment by ptibogxiv** [comment]
@eldy  i will work to add full stripe source's support :

**Comment by ptibogxiv** [comment]
@eldy i updated this commit to get support of more sources but I have pb with dolibarr CSS. 
For test google/ apple pay, you must add  a .well-know/ directory as Stripe's docs

dolibarr

File: htdocs/public/payment/newpayment.php [hunk 2]
```diff
@@ -432,16 +432,36 @@
...
+			//	$card = $stripeToken;
+			//}
+      
+      $pos=strpos($card,'src_');
+
+if ($pos !== false) {
+$src = \Stripe\Source::retrieve("$card",array("stripe_account" => $stripeacc));
+}
+else {
+$src = \Stripe\Source::create(array(
```
Comment (eldy) [inline htdocs/public/payment/newpayment.php:459]: We do not provide the $metadata anymore when creating a card ?

**Pull request closed without merging**